### ISSUE TITLE
Difficult to debug Unexpected indentation errors in `chpldoc`

### ISSUE THREAD

**Issue by rchinmay** [chapel-lang/chapel#19216]
### Summary of Problem

I am a new user to ``chpldoc``, and debugging issues related to unexpected indentation seemed difficult to debug, since the line number printed in the error, would not actually match the line number in the program, and in a program with around 100 comment lines, it was difficult to debug where the error was. The sample source code which was causing this problem is given below. In the source code given below, the error points to line 28 on the ``.rst`` file, which would have been deleted, and line number 28 was also not a valid line number for the source code, since it contains only 26 lines. I tried removing all indentations in comments, but that resulted in an unformatted documentation view. Only after a couple of days of debugging did I find that the actual problem was just a missing empty before listing of argument descriptions. I don't know if it is possible to locate the line number causing the error in the actual source code, or keep the ``.rst`` file (without deleting) as logging for error for easier debugging, but it would actually help for new users with such silly errors in their programs.

### Steps to Reproduce

**Source Code:**

```chapel
/*
  Splits the string according to parsing rules specified and returns the
  list of tokens.
  :arg s: The string to be parsed by shlex.
  :type s: `string`
  :arg comments: Decides whether `'#'` should be considered as a comment
                starting character. Defaults to `false`.
  :type comments: `bool`
  :arg posix: Argument to decide parsing rules (POSIX or non-POSIX). Defaults
              to `false`. Defaults to `true`.
  :type posix: `bool`
  :arg punctuation_chars: If set to true, contiguous runs of punctuation
                          characters will be returned as a single token.
                          Defaults to `false`.
  :type punctuation_chars: `bool`
  :arg custom_punctuation_chars: All the character present in this string
                                will be considered as punctuation characters.
                                NOTE: You can also use this field without
                                setting `punctuation_chars` to true.
                                Defaults to `''` (empty string)
  :type custom_punctuation_chars: `string`
*/
proc split(s: string, comments: bool = false, posix: bool = true,
            punctuation_chars: bool = false,
              custom_punctuation_chars: string = "")
{}
```

**Compile command:**
 `chpldoc Chapel_Program.chpl`

**Compiler Error:**
```
Running Sphinx v4.0.2
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: [new config] 2 added, 0 changed, 0 removed
reading sources... [100%] modules/Chapel_Program

Warning, treated as error:
/tmp/chpldoc-rchinmay.deleteme-cCYkKs/source/modules/Chapel_Program.rst:28:Unexpected indentation.
error: building html output from chpldoc sphinx project
```

### Configuration Information

- Output of `chpl --version`:
```
chpl version 1.26.0 pre-release (d786107f37)
Copyright 2020-2021 Hewlett Packard Enterprise Development LP
Copyright 2004-2019 Cray Inc.
(See LICENSE file for more details)
```
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none *
CHPL_TASKS: fifo *
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: cstdlib *
CHPL_ATOMICS: cstdlib
CHPL_GMP: none *
CHPL_HWLOC: none
CHPL_RE2: bundled *
CHPL_LLVM: none *
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
```
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by lydia-duncan** [comment]
I think this is a duplicate of #16903.  Sorry you're running into this issue :(
